docs/dapp/opl/README.md

```diff
@@ -5,19 +5,147 @@ empowers developers to add new functionality to smart contracts on the most
 popular EVM networks like Ethereum, BNB Chain, and Polygon with encrypted
 transactions and confidential state.
 
-## DApp Development
-
-The Oasis Privacy Layer contains a set of [smart contracts](https://github.com/oasisprotocol/sapphire-paratime/tree/main/contracts/contracts/opl)
-that integrate [Sapphire](../sapphire/README.mdx) into your existing and future
-Web3 applications, introducing confidentiality.
-
-Application developers on any EVM-compatible network can add confidential state
-and selective disclosures to their application through the usage of a smart
-contract deployed on OPL without migrating their application stack. OPL enabled
-smart contracts provide a default bridge for developers to pass messages
-between Sapphire and other networks.
+By using [a Solidity library](https://github.com/oasisprotocol/sapphire-paratime/tree/main/contracts/contracts/opl)
+that integrates [Sapphire](https://oasisprotocol.org/sapphire) into your
+existing and future Web3 applications, developers on any supported network can
+seamlessly add confidential state and selective disclosures to their dApps when
+using contract a deployed on Sapphire without leaving their existing networks.
 
 For more information about OPL and to catch the latest news, please visit the
-official [page](https://oasisprotocol.org/opl).
+[official OPL page](https://oasisprotocol.org/opl).
 
 ![Oasis Privacy Layer diagram](../images/opl/privacy-layer-diagram.png)
+
+The user submits a transaction on the Home network to a contract which uses
+`postMessage` to notify the <abbr title="State Guardian Network">SGN</abbr>
+that it should approve the cross-chain message. The Executor waits, when the SGN
+approves the message the Executor submits a transaction to the target contract
+on Sapphire.
+
+```mermaid
```

Comment:
Diagram source (named `opl-contract-flow.mmd`?) should go here:
https://github.com/oasisprotocol/docs/tree/main/docs/dapp/diagrams

Then, run `yarn diagrams` and Oasis-styled `opl-contract.mmd.svg` should appear in the diagrams folder which you then include as a regular image.

Reply:
I've just removed the diagram as `yarn diagrams` requires headless chrome which is a pain to run inside docker with my current config.